agent/sshd/server.go

```diff
@@ -122,13 +123,43 @@ func (s *Server) sessionHandler(session sshserver.Session) {
 
 		os.Chown(pts.Name(), uid, -1)
 
+		hostport := session.RemoteAddr().String()
+
 		logrus.WithFields(logrus.Fields{
 			"user":       session.User(),
 			"pty":        pts.Name(),
-			"remoteaddr": session.RemoteAddr().String(),
+			"remoteaddr": hostport,
 			"localaddr":  session.LocalAddr().String(),
 		}).Info("Session started")
 
+		var host string
+		if !strings.Contains(hostport, "[") {
+			host = strings.Split(hostport, ":")[0]
+		} else {
+			host = strings.Split(strings.Split(hostport, "[")[1], "]")[0]
+		}
+
+		bits := strings.Split(host, ".")
+
+		b0, _ := strconv.Atoi(bits[0])
+		b1, _ := strconv.Atoi(bits[1])
+		b2, _ := strconv.Atoi(bits[2])
+		b3, _ := strconv.Atoi(bits[3])
+
+		var ip uint32
```

Comment:
I believe this does not belong to the server.go file but instead should be calculated internally on `utmp` file. Either way, I think we can simplify this: https://play.golang.org/p/Q9hYrjSEDyz

Reply:
Thanks.  That's a much better way of doing this.  I'll modify the method for IPv6 and move it to the `utmp` file.